src/m365/spo/commands/page/page-set.ts

```diff
@@ -209,6 +211,11 @@ class SpoPageSetCommand extends SpoCommand {
           if (authorByline) {
             pageData.AuthorByline = authorByline;
           }
+
+          // Needs to be at the end, as the data is still needed in the last step
+          if (!needsToSavePage) {
```

Comment:
Won't this prevent the page from being saved unless you updated title or description? It seems like there are other properties in that section that can be updated as well.

Reply:
Nope, because the last step is to save it or publish it.

Reply:
If that's being done at the end, do we need to do it here at all?

Reply:
So if you only want to publish the page, without any change. You do not need to save the page in between. 

If you would change the title and publish, you will have to save the page and publish it.

If you want to update the title, but not publish. You save the page, and save it as draft in the end to get it check in.

Reply:
What's unclear to me is why this is tied to promoting the page rather than the publish flag and why the check applies only when `promoteAs` is different than `Template` which doesn't seem to be related to publishing itself.

Reply:
IMO there is too much logic in this command. I would have separated the page and template logic in general from the command. This is of course still something we can do but would be for a new version as it comes with a breaking change.

But the logic is, that if the `promoteAs` is not a template, it will be a regular page and all properties need to be passed when saving it, a template doesn't need this. Cannot be published.

So in the end, the only thing that this change implement is the extra page save before publishing.

Reply:
With the breaking change, do you mean you'd suggest breaking it up in two commands? Couldn't we achieve the same in one? Having too granular commands complicates discoverability.

Reply:
You can, as that is how it is right now. Managing templates is in my opinion a bit different. 

I'll do an analysis later.

Reply:
I just saw we have a command for listing page templates and @appieschot just added a spec for removing page templates. Since we're going that direction already, we could introduce new commands for adding and updating page templates and deprecate the existing functionality in this command which we'd remove in v4 later this year.

Reply:
Happy to chat and align our thoughts.